node/src/node_configurator/mod.rs

```diff
@@ -11,34 +11,218 @@ use crate::database::db_initializer::{DbInitializer, DbInitializerReal};
+    pub(crate) real_user_spec: bool,
+    pub(crate) data_directory: PathBuf,
+    pub(crate) data_directory_spec: bool,
+    pub(crate) config_file: PathBuf,
+    pub(crate) config_file_spec: bool,
+}
+
+fn get_chain_from_vcl(configs: &[Box<dyn VclArg>]) -> (Chain, bool) {
+    match argument_from_vcl(configs, "--chain") {
+        Some(chain) => (Chain::from(chain), true),
+        None => (DEFAULT_CHAIN, false),
+    }
+}
+
+fn get_real_user_from_vcl(
+    configs: &[Box<dyn VclArg>],
+    dirs_wrapper: &dyn DirsWrapper,
+) -> (RealUser, bool) {
+    match argument_from_vcl(configs, "--real-user") {
+        Some(user) => {
+            let real_user_split: Vec<&str> = user.split(':').collect();
+            (
+                RealUser::new(
+                    Some(real_user_split[0].parse::<i32>().expect("expected user id")),
+                    Some(
+                        real_user_split[1]
+                            .parse::<i32>()
+                            .expect("expected user group"),
+                    ),
+                    Some(PathBuf::from(real_user_split[2])),
+                ),
+                true,
+            )
+        }
+        None => {
+            #[cfg(target_os = "windows")]
+            {
+                (
+                    RealUser::new(Some(999999), Some(999999), None).populate(dirs_wrapper),
+                    false,
+                )
+            }
+            #[cfg(not(target_os = "windows"))]
+            {
+                (
+                    RealUser::new(None, None, None).populate(dirs_wrapper),
+                    false,
+                )
+            }
+        }
+    }
+}
+
+fn get_data_directory_from_vcl(
+    configs: &[Box<dyn VclArg>],
+    dirs_wrapper: &dyn DirsWrapper,
+    real_user: &RealUser,
+    chain: &Chain,
+) -> (PathBuf, bool) {
+    match argument_from_vcl(configs, "--data-directory") {
+        Some(data_dir) => match PathBuf::from(data_dir).starts_with("~/") {
+            true => {
+                let home_dir_from_wrapper = dirs_wrapper
+                    .home_dir()
+                    .expect("expexted users home dir")
+                    .to_str()
+                    .expect("expect home dir")
+                    .to_string();
+                let replaced_tilde_dir =
+                    data_dir
+                        .to_string()
+                        .replacen('~', home_dir_from_wrapper.as_str(), 1);
+                (PathBuf::from(replaced_tilde_dir), true)
+            }
+            false => (PathBuf::from(&data_dir), true),
+        },
+        None => (
+            data_directory_from_context(dirs_wrapper, real_user, *chain),
+            false,
+        ),
+    }
+}
+
+fn get_config_file_from_vcl(
+    configs: &[Box<dyn VclArg>],
+    data_directory: &PathBuf,
+    data_directory_def: bool,
+    dirs_wrapper: &dyn DirsWrapper,
+) -> (PathBuf, bool) {
+    match argument_from_vcl(configs, "--config-file") {
```

Comment:
We make a PathBuf from this string many times below; perhaps we should make the PathBuf once at the top and use it many times instead?

Reply:
Also...what if the string in `--config-file` actually names a directory? Is that handled here?